test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/BiPredicateTest.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.function.BiPredicate
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class BiPredicateTest {
+  import BiPredicateTest._
+
+  private val largerThan10LessThan30: BiPredicate[Int, Int] = makeBiPredicate { (t, u) => t > 10 && u < 30 }
+  private val even: BiPredicate[Int, Int] = makeBiPredicate { (t, u) => isEven(t) && isEven(u) }
+
+  private val throwingPredicate: BiPredicate[Int, Int] = makeBiPredicate { (t, _) =>
+    throw new ThrowingPredicateException(t)
+  }
+
+  private val dontCallPredicate: BiPredicate[Int, Int] = makeBiPredicate { (t, u) =>
+    throw new AssertionError(s"dontCallPredicate.test($t, $u)")
+  }
+
+  @Test def and(): Unit = {
+    val evenAndInRange = largerThan10LessThan30.and(even)
+
+    assertTrue("Both In Range and Even", evenAndInRange.test(22, 22))
+    assertFalse("Both In Range and Not Even", evenAndInRange.test(41, 22))
+    assertFalse("One In Range and Even", evenAndInRange.test(6, 6))
+    assertFalse("One In Range, One Not Even", evenAndInRange.test(21, 42))
```

Comment:
This still tests the implementation in the test. Please fix this everywhere. I'm not going to point out every spot.

Reply:
I think this represents the truth table for testing `and`, maybe comments are wrong?

Would these comments clarify instead?

// Left true, Right True
// Left true, Right False
// Left false, Right true
// Left false, Right false

Or maybe I don't understand what you mean about the "implementation"

Reply:
This seems to do the right thing to me. It tests that `BiPredicate.and()` correctly implements the truth table of its two components. This is similar to what we have in the test of `Predicate`:
https://github.com/scala-js/scala-js/blob/49dd4263301fe9d099906ccd02c0b4f27e479591/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/PredicateTest.scala#L34-L46

Reply:
I can remove the `largerThan10LessThan30` from here, perhaps that is it, and simplify that

Reply:
Ah, yes. Indeed. Apologies. The truth table is correct, it's just the comments that tripped me off. (The comments that are there now are good IMO).